lib/hocon/impl/abstract_config_object.rb

```diff
@@ -83,6 +83,10 @@ def peek_path(path)
     end
   end
 
+  def resolve_substitutions(context, source)
+    raise "'resolve_substitutions' needs to be overridden by sub-class, this is abstract in the Java version"
```

Comment:
Might be good to add a TODO here so we remember to come back and implement this.